builtin/logical/database/backend.go

```diff
@@ -58,15 +58,16 @@ func Factory(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 	// Create a context with a cancel method for processing any WAL entries and
 	// populating the queue
 	initCtx := context.Background()
-	ictx, cancel := context.WithCancel(initCtx)
-	b.cancelQueue = cancel
+	b.ctx, b.cancelQueue = context.WithCancel(initCtx)
```

Comment:
Since we're always setting this within `Backend`, do we have to set it in `Factory` at all?

Reply:
It is still necessary if we want to tie the backend context to the passed in context, but I think I can refactor it a little to simplify it.